prog/github/github_repository_nexus.rb

```diff
@@ -91,6 +91,18 @@ def cleanup_cache
       .limit(200)
       .destroy
 
+    # Destroy cache entries if it is created 20 minutes ago
+    # but couldn't committed yet. 20 minutes decided as during
+    # our performance tests uploading 10GB of data (which is
+    # the max size for a single cache entry) takes ~8 minutes at most.
+    # To be on the safe side, 1.5x buffer is added.
+    twenty_minutes_ago = Time.now - 20 * 60
```

Comment:
In the worst case, if it uploads at 10 MB/s, it will take 17 minutes. Let's be on the safer side and increase it to 30 minutes.